src/lang-go.ts

```diff
@@ -727,63 +660,149 @@ export async function activateUsingWebSockets(ctx: sourcegraph.ExtensionContext)
+                requestType: lsp.HoverRequest.type,
+                request: positionParams(doc, pos),
+                useCache: true,
+            })
+        )
+
+    const definition = async (doc: sourcegraph.TextDocument, pos: sourcegraph.Position) =>
+        convert.xdefinition({
+            currentDocURI: doc.uri,
+            xdefinition: await sendRequest<lspext.Xdefinition[] | null>({
+                rootURI: rootURIFromDoc(doc),
+                requestType: new lsp.RequestType<any, any, any, void>('textDocument/xdefinition') as any,
+                request: positionParams(doc, pos),
+                useCache: true,
+            }),
+        })
+
+    const references = async (doc: sourcegraph.TextDocument, pos: sourcegraph.Position) =>
+        convert.references({
+            currentDocURI: doc.uri,
+            references: await sendRequest<LSP.Location[]>({
+                rootURI: rootURIFromDoc(doc),
+                requestType: lsp.ReferencesRequest.type,
+                request: positionParams(doc, pos),
+                useCache: true,
+            }),
+        })
+
+    registerExternalReferences({ ctx, sendRequest, settings })
+    registerImplementations({ ctx, sendRequest })
+
+    return {
+        hover: wrapMaybe(hover),
+        definition: wrapMaybe(definition),
+        references: wrapMaybe(references),
+    }
+}
+
 function pathname(url: string): string {
     let pathname = url
     pathname = pathname.slice('git://'.length)
     pathname = pathname.slice(0, pathname.indexOf('?'))
     return pathname
 }
 
+function initBasicCodeIntel() {
+    const handler = new Handler({
+        sourcegraph,
+        languageID: 'go',
+        fileExts: ['go'],
+        filterDefinitions: ({ repo, filePath, pos, fileContent, results }) => {
+            const currentFileImportedPaths = fileContent
+                .split('\n')
+                .map(line => {
+                    // Matches the import at index 3
+                    const match = /^(import |\t)(\w+ |\. )?"(.*)"$/.exec(line)
+                    return match ? match[3] : undefined
+                })
+                .filter((x): x is string => Boolean(x))
+
+            const currentFileImportPath = repo + '/' + path.dirname(filePath)
+
+            const filteredResults = results.filter(result => {
+                const resultImportPath = result.repo + '/' + path.dirname(result.file)
+                return (
+                    currentFileImportedPaths.some(i => resultImportPath.includes(i)) ||
+                    resultImportPath === currentFileImportPath
+                )
+            })
+
+            return filteredResults.length === 0 ? results : filteredResults
+        },
+        commentStyle: {
+            lineRegex: /\/\/\s?/,
+        },
+    })
+    return {
+        hover: handler.hover.bind(handler),
+        definition: handler.definition.bind(handler),
+        references: handler.references.bind(handler),
+    }
+}
```

Comment:
What's inside `initBasicCodeIntel` already existed, I just made a new function out of it.